Preflight checks fail when directories are not present

### Observed behavior

When using the [simple.json](https://github.com/synadia-io/nex/blob/main/examples/nodeconfigs/simple.json) preflight checks, the command fails if the directories are not already present.

### Expected behavior

Preflight checks should create any required nested directories.

### Nex and NATS version

```bash
root@nats-client:~# nex --version
vdevelopment [] | Built-on: 
```

### Host environment

Ubuntu 23.10 x64

### Steps to reproduce

I'm using a clean Ubuntu 23.10 x64 droplet in Digital Ocean, with golang and the nats CLI installed.

1. Install nex cli
2. Save https://github.com/synadia-io/nex/blob/main/examples/nodeconfigs/simple.json to your local directory
```bash
root@nats-client:~# nex node preflight --config simple.json 
Validating - Required CNI Plugins [/opt/cni/bin]
	⛔ Missing Dependency - /opt/cni/bin/host-local [host-local CNI plugin]
	⛔ Missing Dependency - /opt/cni/bin/ptp [ptp CNI plugin]
	⛔ Missing Dependency - /opt/cni/bin/tc-redirect-tap [tc-redirect-tap CNI plugin]

Validating - Required binaries [/usr/local/bin]
	⛔ Missing Dependency - /usr/local/bin/firecracker [Firecracker VM binary]

Validating - CNI configuration requirements [/etc/cni/conf.d]
	⛔ Missing Dependency - /etc/cni/conf.d/fcnet.conflist [CNI Configuration]

Validating - User provided files []
	⛔ Missing Dependency - /tmp/wd/vmlinux [VMLinux Kernel]
	⛔ Missing Dependency - /tmp/wd/rootfs.ext4 [Root Filesystem Template]

⛔ You are missing required dependencies for [Required CNI Plugins], do you want to install? [y/N] y
open /opt/cni/bin/ptp: no such file or directory
panic: preflight checks failed: open /opt/cni/bin/ptp: no such file or directory

goroutine 1 [running]:
github.com/synadia-io/nex/internal/node.CmdPreflight(0x1114db0?, 0x1784a80, {0xe58680?, 0xc0003fab40?}, 0xc0003fab40?, 0xc00046c5f0?)
	/root/go/pkg/mod/github.com/synadia-io/nex@v0.0.0-20240122212631-fd8d975937bb/internal/node/init.go:140 +0x96
main.RunNodePreflight(0x40e44c?)
	/root/go/pkg/mod/github.com/synadia-io/nex@v0.0.0-20240122212631-fd8d975937bb/nex/node_linux_amd64.go:33 +0x179
github.com/choria-io/fisk.(*actionMixin).applyActions(...)
	/root/go/pkg/mod/github.com/choria-io/fisk@v0.6.1/actions.go:28
github.com/choria-io/fisk.(*Application).applyActions(0xc00047f400?, 0xc00019e000)
	/root/go/pkg/mod/github.com/choria-io/fisk@v0.6.1/app.go:823 +0xdf
github.com/choria-io/fisk.(*Application).execute(0xc00047f400, 0xc00019e000?, {0xc0003fab80, 0x2, 0x2})
	/root/go/pkg/mod/github.com/choria-io/fisk@v0.6.1/app.go:609 +0x65
github.com/choria-io/fisk.(*Application).Parse(0xc00047f400, {0xc000100060?, 0xc0004eb650?, 0x30?})
	/root/go/pkg/mod/github.com/choria-io/fisk@v0.6.1/app.go:272 +0x16c
github.com/choria-io/fisk.(*Application).MustParseWithUsage(0xc00047f400, {0xc000100060, 0x4, 0x4})
	/root/go/pkg/mod/github.com/choria-io/fisk@v0.6.1/app.go:889 +0x45
main.main()
	/root/go/pkg/mod/github.com/synadia-io/nex@v0.0.0-20240122212631-fd8d975937bb/nex/nex.go:94 +0x41f7
```

If you run `mkdir /opt/cni/bin` and run again, the CNI Plugins step succeeds but then fails on `CNI configuration requirements`. After resolving that issue, it fails again at `User provided files`. After creating all required directories, preflight succeeds.

--- a/internal/node/prereq.go
+++ b/internal/node/prereq.go
@@ -143,7 +143,20 @@ func CheckPreRequisites(config *NodeConfiguration) error {
 				return err
 			}
 		}
-		if config.ForceDepInstall || string(input) == "y\n" {
+		if config.ForceDepInstall || strings.ToUpper(string(input)) == "Y\n" {
+			var path string
+			for _, f := range r.files {
+				if r.directory == "" {
+					path = filepath.Dir(f.name)
+				} else {
+					path = r.directory
+				}
+				err = os.MkdirAll(path, 0755)
+				if err != nil {
+					return err
+				}
+			}
+
 			for _, iF := range r.initFuncs {
 				err := iF(r, config)
 				if err != nil {

